packages/back-end/src/routers/project/project.controller.ts

```diff
@@ -197,7 +197,9 @@ export const deleteProject = async (
   // Clean up metrics
   if (deleteMetrics) {
     try {
-      req.checkPermissions("createAnalyses", id);
+      if (!context.permissions.canDeleteMetric({ projects: [id] })) {
+        context.permissions.throwPermissionError();
+      }
```

Comment:
Is this change intentional?  Before it was checking `createAnalyses`, but now it's checking metric permissions instead.

Reply:
It was, but it's possible I'm missing something.

`createAnalyses` seems to relate to `experiments` and `ideas`, not necessarily directly with `metrics`.

Maybe we want to update `canDeleteMetric` to get a list of experiments using the metric, and ensure the user has `createAnalyses` on all of those experiments to protect a user from deleting a metric associated with an experiment they don't have permission to impact.

If we do that check specifically, we could then customize the error message they receive. It might be confusing to get a simple "You don't have permission", and it might be more helpful to get a "This metric is associated with an experiment you don't have permission to access, so you can not delete it."

Reply:
I think your change is fine. Let's leave it